runtime/parachains/src/ump.rs

```diff
@@ -207,44 +203,36 @@ impl<T: Config> Module<T> {
 	/// Called by the initializer to note that a new session has started.
 	pub(crate) fn initializer_on_new_session(
 		_notification: &initializer::SessionChangeNotification<T::BlockNumber>,
+		outgoing_paras: &[ParaId],
 	) {
-		Self::perform_outgoing_para_cleanup();
+		Self::perform_outgoing_para_cleanup(outgoing_paras);
 	}
 
-	/// Iterate over all paras that were registered for offboarding and remove all the data
+	/// Iterate over all paras that were noted for offboarding and remove all the data
 	/// associated with them.
-	fn perform_outgoing_para_cleanup() {
-		let outgoing = OutgoingParas::take();
+	fn perform_outgoing_para_cleanup(outgoing: &[ParaId]) {
 		for outgoing_para in outgoing {
 			Self::clean_ump_after_outgoing(outgoing_para);
 		}
 	}
 
-	/// Schedule a para to be cleaned up at the start of the next session.
-	pub(crate) fn schedule_para_cleanup(id: ParaId) {
-		OutgoingParas::mutate(|v| {
-			if let Err(i) = v.binary_search(&id) {
-				v.insert(i, id);
-			}
-		});
-	}
-
-	fn clean_ump_after_outgoing(outgoing_para: ParaId) {
-		<Self as Store>::RelayDispatchQueueSize::remove(&outgoing_para);
-		<Self as Store>::RelayDispatchQueues::remove(&outgoing_para);
+	/// Remove all relevant storage items for an outgoing parachain.
+	fn clean_ump_after_outgoing(outgoing_para: &ParaId) {
```

Comment:
Nit: not strictly necessary to change to a shared reference: `ParaId` is a copy type

Reply:
is there a reason to not be a shared reference? This would be the natural type coming from a slice.

Reply:
it would be natural either way: again because it's a `Copy` type. You can think of it as similar to a `u32`, so it's always possible to "move it out" from the slice by basically making a copy.

FWIW, on a lower lever, I am pretty sure there is no difference because LLVM will be able to note that using a pointer doesn't make sense because, the pointer requires a dereference and twice as big as the value itself.

↑ But that's a nit, feel free to ignore anyway